### PR TITLE
chore: Add release verification script

### PR DESCRIPTION
**Pull request by robin-aws**
Adds a utility script to verify that a released package can be at least installed and loaded, along with a top-level npm script to run this for each package.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: util/verify_release [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// walk the file tree upwards and potentially find unrelated node_modules directories!
+// TODO-RS: An empty directory triggers several warnings, about things like missing
+// package.json and package-lock.json files. Can we clean this up in the future?
+const consumingPackageDir = tmp.dirSync();
+
+// Manually install the dependency
+execSync(`npm install --prefix ${consumingPackageDir.name} ${releaseIdentifier}`)
+
+// Verify that it can be imported. This also ensures that types are available correctly.
+execSync(`ts-node -e "import * as everything from '${packageName}'"`)
```
Comment (seebees) [inline util/verify_release:34]: For further improvements:

The package.json has an optional section [files](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files) that will control what files are actually published. This is generally a good practice. If we add this section to these packages, then we can further enhance this script.

We could create a file in each package `test/verify_publish.ts`. To start each file could just import as you have above. But as we find additional things to test, this is a nice place to put such things. Obviously browser vs node environments might frustrate us, but this is a good start.

File: util/verify_release [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+// A very simple script to verify that a package can be installed and imported.
+// Intended primarily as a basic smoke test after publishing.
+// TODO-RS: Work out how to run this against verdaccio for pre-release verification too!
+
+const { execSync } = require('child_process')
+const path = require('path');
+const tmp = require('tmp')
+var fs = require('fs');
```
Comment (seebees) [inline util/verify_release:12]:
```suggestion
const fs = require('fs');
```

**Review by seebees — APPROVED** [review]

File: util/verify_release (hunk 1 above)
Comment (seebees) [reply util/verify_release:34]: To be clear, then I think you cp the file and run it in the temp dir...

Comment (robin-aws) [reply util/verify_release:34]: I'm pretty sure I don't like the idea of putting anything that tests the package "from the outside" in the package itself. That always leads to confusion and subtle encapsulation bugs in testing. I think the JS ESDK does the right thing of defining separate `integration-*` packages, and if we want deeper testing we should go down that road instead IMO.

**Review by seebees — APPROVED** [review]

**Pull request merged**